url913d8566d1_AnsBalin_rbn_tgz-pJ8-MainJ8/src/Network.java

```diff
@@ -0,0 +1,657 @@
+import java.io.*;
+import java.util.*;
+
+public class Network implements Serializable{
+  
+  private int numNodes;
+  private int NetworkID;
+  private int population;
+  private int numBondingSites;
+  private ArrayList<Node> nodes = new ArrayList<Node>();
+  private int[] bondingSites;
+  
+  private int state;  
+  private double activity;
+  private int cycleLength;
+  
+  private Molecule mol;
+  
+  public Network(){
+    
+    numNodes=0;
+    NetworkID=0;
+    numBondingSites=0;
+    nodes = null;
+    bondingSites = new int[0];
+    
+  }
+  
+  // Constructor for a NON-bonding RBN
+  // If numNodes = 0, constructs a NULL NETWORK
+  public Network(int numNodes){
+    
+    this.numNodes = numNodes;
+    this.NetworkID = 0;
+    this.bondingSites = new int[1];
+    
+    for(int i=0;i<numNodes;i++){
+      
+      nodes.add(new Node(numNodes, false)); //currently set to create non bonding RBNs
+    }
+    
+    calculateCycleLength();
+    calculateActivity();
+  }
+  
+  // Constructor for a bonding RBN: Same as above, but nodes are ordered by connectivity
+  // and most highly ordered nodes are designated as bonding sites
+  public Network(int numNodes, int NetworkID, int numBondingSites){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 60-103 into a separate method.